lib/3scale/backend/listener_metrics.rb

```diff
@@ -13,19 +13,21 @@ class ListenerMetrics
       }
       private_constant :AUTH_AND_REPORT_REQUEST_TYPES
 
+      # Only the first match is taken into account, that's why for example,
+      # "/\/services\/.*\/stats/" needs to appear before "/\/services/"
```

Comment:
We could write some code to ensure this gets sorted properly regardless of people ignoring this comment and breaking it. Probably not worth the effort, though... unless you start automating the information about these endpoints and generate also their associated methods for building paths usable in tests, etc.

Reply:
I thought about that but I don't think it's worth the effort. Also, now we have tests that should fail if this is not sorted.